connexion/decorators/uri_parsing.py

```diff
@@ -96,8 +97,22 @@ def resolve_params(self, params, _in):
         """
         resolved_param = {}
         for k, values in params.items():
+            groups = fullmatch(r'^(\w+)\[{1}(\w+)\]{1}$', k)
+            if groups:
+                possible_key = groups.group(1)
+                param_defn = self.param_defns.get(possible_key)
+                if param_defn \
+                        and param_defn.get('style', None) == 'deepObject' and param_defn.get('explode', False):
```

Comment:
check out `_resolve_parameter_duplicates(...)` and `_split(...)`
these should get rid of the array unwrapping that you are doing in other places.

Reply:
not sure what you mean with that. but i moved the array unnesting now to `uri_parsing`. would that be sufficient enough?